### ISSUE TITLE
API Error. Request could not be processed.

### ISSUE THREAD

**Issue by agk1190** [Hyundai-Kia-Connect/kia_uvo#611]
**Region and Brand of car**
2023 Kia Telluride X-Line
Canada

**Version of the integration**
v2.6.1

**Describe the bug**
Triggering an update gives the error:
`hyundai_kia_connect_api.exceptions.APIError: Server returned: 'We're sorry, but your request could not be processed. Please try again. If the issue persists call Customer assistance at 1-877-542-2886.'`

This occurs when trying to do either a soft update or a force update but only after an update has run after each reboot.
In other words, it seems that I can only call the update service once before having to reboot Home Assistant. Any subsequent update calls without rebooting will fail.

**Debug logs if an error occurred**

```
Logger: homeassistant.components.automation.kia_update
Source: custom_components/kia_uvo/coordinator.py:166
Integration: Automation (documentation, issues)
First occurred: March 13, 2023 at 11:00:00 PM (2 occurrences)
Last logged: March 13, 2023 at 11:00:00 PM

Kia Update: Error executing script. Unexpected error for call_service at pos 1: Server returned: 'We're sorry, but your request could not be processed. Please try again. If the issue persists call Customer assistance at 1-877-542-2886.'
While executing automation automation.kia_update
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 451, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 684, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1808, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1845, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/config/custom_components/kia_uvo/services.py", line 54, in async_handle_update
    await coordinator.async_update_all()
  File "/config/custom_components/kia_uvo/coordinator.py", line 150, in async_update_all
    await self.async_check_and_refresh_token()
  File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 121, in check_and_refresh_token
    self.token: Token = self.api.login(self.username, self.password)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiCA.py", line 127, in login
    self._check_response_for_errors(response)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiCA.py", line 117, in _check_response_for_errors
    raise APIError(f"Server returned: '{response['error']['errorDesc']}'")
hyundai_kia_connect_api.exceptions.APIError: Server returned: 'We're sorry, but your request could not be processed. Please try again. If the issue persists call Customer assistance at 1-877-542-2886.'
```

**Expected behavior**
Should be able to update / force update multiple times without having to reboot Home Assistant.

Is this likely a Kia error or an integration error?

**Comment by agk1190** [comment]
Update: Updating to Home Assistant Core v2023.3.4 may have solved this issue. Still need to test some more.

**Comment by agk1190** [comment]
Update 2: Issue still persists.

**Comment by cdnninja** [comment]
I'll see if I can get some time to try replicate tomorrow.

**Comment by cdnninja** [comment]
Does the native app work for update calls for you?

**Comment by cdnninja** [comment]
I can't seem to replicate on my Canadian 2019 kia sorento.

**Comment by agk1190** [comment]
Yes, the Kia Connect app works fine.

I just tried to do another update call through Home Assistant and it failed with the same error. 
I then restarted Home Assistant and it was able to do an update on startup. I was also able to do a force update successfully.

I will post an update if/when it stops working again.

**Comment by adorobis** [comment]
I'm facing the same issue now:
```
2023-03-20 18:25:18.898 ERROR (SyncWorker_0) [hyundai_kia_connect_api.KiaUvoApiEU] Unknown API response format: {'error': 'There was a problem proxying the request'}
2023-03-20 18:25:18.927 ERROR (MainThread) [custom_components.kia_uvo.coordinator] Unexpected error fetching kia_uvo data: 
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 239, in _async_refresh
    self.data = await self._async_update_data()
  File "/config/custom_components/kia_uvo/coordinator.py", line 107, in _async_update_data
    await self.async_check_and_refresh_token()
  File "/config/custom_components/kia_uvo/coordinator.py", line 166, in async_check_and_refresh_token
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 118, in check_and_refresh_token
    self.initialize()
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 62, in initialize
    self.token: Token = self.api.login(self.username, self.password)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 198, in login
    device_id = self._get_device_id(stamp)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1103, in _get_device_id
    _check_response_for_errors(response)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 104, in _check_response_for_errors
    raise InvalidAPIResponseError()
hyundai_kia_connect_api.exceptions.InvalidAPIResponseError
```
Running latest HA version (2023.3.5) and latest Kia integration.

**Comment by agk1190** [comment]
So I completely reinstalled the integration in Home Assistant a few days after my last comment and left everything at the default values except the Force Refresh Interval which I set to 360min. I also left the Enable Polling for Updates option enabled (which I had disabled before). 

Everything seemed to be working fine (updating every so often) for a few days until it stopped. The app also stopped working at this point. I called Kia and they had me reset the something in the car. They did something on their end as well. This fixed the issue with both the app and the integration. 

A few days ago, the integration started failing randomly and all entities would show up as 'Unavailable'. 
![image](https://user-images.githubusercontent.com/90426659/230702612-7b4b086d-3cce-42ab-8524-623ee51c6856.png)

Error message:
```
Logger: custom_components.kia_uvo.coordinator
Source: custom_components/kia_uvo/coordinator.py:126
Integration: Hyundai / Kia Connect (documentation, issues)
First occurred: 7:44:00 AM (10 occurrences)
Last logged: 5:02:10 PM

Force update failed, falling back to cached: 'result'
Traceback (most recent call last):
  File "/config/custom_components/kia_uvo/coordinator.py", line 126, in _async_update_data
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 99, in check_and_force_update_vehicles
    self.force_refresh_vehicle_state(vehicle_id)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/VehicleManager.py", line 112, in force_refresh_vehicle_state
    self.api.force_refresh_vehicle_state(self.token, vehicle)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiCA.py", line 197, in force_refresh_vehicle_state
    state = self._get_forced_vehicle_state(token, vehicle)
  File "/usr/local/lib/python3.10/site-packages/hyundai_kia_connect_api/KiaUvoApiCA.py", line 470, in _get_forced_vehicle_state
    response = response["result"]["status"]
KeyError: 'result'
```


It would stay like this until I reloaded the integration. The last 6 days of timeline is shown below. The gaps are when the integration was unavailable:
![image](https://user-images.githubusercontent.com/90426659/230702676-e63cc552-0058-4c2f-9b54-56e137fd7936.png)


And finally, last night (Apr 6), the integration stopped working but now with the following error:
```
Logger: custom_components.kia_uvo.coordinator
Source: helpers/update_coordinator.py:193
Integration: Hyundai / Kia Connect (documentation, issues)
First occurred: April 6, 2023 at 11:31:28 PM (1 occurrences)
Last logged: April 6, 2023 at 11:31:28 PM

Error requesting kia_uvo data: HTTPSConnectionPool(host='kiaconnect.ca', port=443): Max retries exceeded with url: /tods/api/lstvhclsts (Caused by SSLError(SSLError(1, '[SSL: WRONG_SIGNATURE_TYPE] wrong signature type (_ssl.c:997)')))
```
This seems to be the same issue as #600 a month ago.

The integration does not load now but the app still works.
Again, any help is appreciated.

**Comment by cdnninja** [comment]
Looks like SSL changed. I have another issue for this but my broke too.  I will take a look soon.

**Comment by cdnninja** [comment]
Just released an update that helps with the SSL item.  Let me know how it goes.  It got me back in but haven't tested all services.

**Comment by agk1190** [comment]
How can I get Home Assistant to download the latest code without you creating a new release?

**Comment by keithajmani** [comment]
@agk1190  — you can just edit the file yourself, it’s a tiny tweak, and restart HA. 

https://github.com/Hyundai-Kia-Connect/kia_uvo/issues/623#issuecomment-1501243849

Cheers.

**Comment by agk1190** [comment]
Thanks @keithajmani for the suggestion. The integration is back up and running now.

**Comment by cdnninja** [comment]
Oops on the release front.  Fixed now - in the future you can use hacs to update and select a version of "master".  Glad it is fixed.
